projects/avodata/firstStage.py

```diff
@@ -0,0 +1,121 @@
+import numpy as np
+import math
+
+
+def factorial(n):
+    return int(math.gamma(n + 1))
+
+
+def n_choose_k(n, k):
+    return factorial(n) / (factorial(k) * factorial(n - k))
+
+
+def generate_synthetic_a(num_of_professions):
+    num_of_rows = int(n_choose_k(num_of_professions, 2))
+    A = np.zeros(shape=(num_of_rows, num_of_professions), dtype=np.float32)
+    row = 0
+
+    for i in range(num_of_professions):
+        for j in range(i + 1, num_of_professions):
+            A[row][i] = 1
+            A[row][j] = -1
+            row += 1
+
+    return A
+
+
+def generate_synthetic_c(m, i, a):
+    c = np.dot(a, m[i].transpose())
+    return c
+
+
+def solve(A, c):
+    row_in_m = np.linalg.lstsq(A, c, None)[0]
+    return row_in_m
+
+def build_m(num_of_professions, c_matrix, num_of_characteristics):
+    A = generate_synthetic_a(num_of_professions)
+    m = np.zeros((num_of_characteristics, num_of_professions), dtype=np.float32)
+    for i in range(num_of_characteristics):
+        c = c_matrix[i]
+        row_in_m = solve(A, c)
+        row_in_m = (row_in_m - np.min(row_in_m)) / np.ptp(row_in_m)
```

Comment:
`ptp` is just `max` in this case, since you subtract the minimum value, so the smallest value is always 0.
So I would just replace it with `max`.
Also, please comment before:
`// We normalize to the [0,1] range:

Reply:
If I just change to max the result is not the same. Because the smallest value is 0 just after all this action actually. So max and ptp are not the same here. 
It's possible with two rows, and i did it... 
I hope its good like this.